pkg/controller/volume/persistentvolume/index.go

```diff
@@ -138,9 +164,24 @@ func findMatchingVolume(claim *v1.PersistentVolumeClaim, volumes []*v1.Persisten
 			if volumeQty.Cmp(requestedQty) < 0 {
 				continue
 			}
+			if node != nil {
+				// Scheduler path, check that the PV NodeAffinity
+				// is satisfied by the node
+				err := volumeutil.CheckNodeAffinity(volume, node.Labels)
+				if err != nil {
+					return nil, nil
```

Comment:
Should the scheduler find a better PV in this case (i.e. `continue`)?

Reply:
ah good catch, will also add a test case